kv/union_store.go

```diff
@@ -77,6 +71,26 @@ func NewUnionStore(snapshot Snapshot) UnionStore {
 	}
 }
 
+type invalidIterator struct{}
```

Comment:
`Seek` returns an `Iterator`, when a transaction is read-only, since no data is written, the iterator is invalid, we don't need to create a MemBuffer to create the iterator.